Using tofudl on tfcache tests and removing incompatible tests

diofeher opened the pull request:
Using `tofudl` instead of `github.com/hashicorp/hc-install/releases` on the tfcache for tofu-exec tests.

This makes sure we're downloading the `tofu` binary when running the tests for different `tofu` versions. This is foundation work to create tests for different versions.

The same unit tests are broken on `main`, but we shouldn't worry about them now since they are testing older versions of Terraform. These tests can be removed later.

Yantrio commented:
Hi, Thanks for this contribution. Right now we are not sure if we want to maintain this repository/fork or not because of the maintainance cost included with that. However right now I am happy to merge this as it's an extremely simple change. But I do recommend that you continue to use terraform-json if possible in the meantime. 

If/When we do make a choice to maintain this package, I will let you know.

diofeher committed:
adding tofudl to go.mod

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Fix tfcache making sure tofu binary is returned for tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

Gogotchuri committed:
Completely remove hc-install from the repository.
We also removed Compatibility test that wasn't relevant and a GitRef method which was highly dependent on hc-install

Signed-off-by: Ilia Gogotchuri <ilia.gogotchuri0@gmail.com>

Gogotchuri commented:
I've added one more commit, to completely get rid of hc-install references and usage from the repository

diofeher committed:
Fix panics in e2e tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Fix e2e tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Remove v0.13 tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Remove upgrade v0.13 tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Fix matrixtest

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Remove v0.14 and earlier tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

diofeher committed:
Remove v0.15 tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>

Gogotchuri committed:
Merge remote-tracking branch 'origin/tofudl' into tofudl

diofeher committed:
Remove v0.13 and v0.14 tests

Signed-off-by: Diogenes Fernandes <diofeher@gmail.com>
